flowzone.yml

```
@@ -1486,12 +1489,26 @@ jobs:
           org: ${{ github.repository_owner }}
           default_branch: ${{ github.event.repository.default_branch }}
           url: https://${{ inputs.cloudflare_website }}.pages.dev
+      - name: Decide what branch name to send CF
+        id: cf_branch
+        run: |
+          CF_BRANCH=""
+          echo ${{ fromJSON(needs.event_types.outputs.pr_merged) }}
+          if [[ ${{ fromJSON(needs.event_types.outputs.pr_merged) }} == 'true' ]]
+          then
+            CF_BRANCH=${{ github.event.repository.default_branch }}
+          else
+            CF_BRANCH=${{ github.event.pull_request.head.ref }}
+          fi
+          echo $CF_BRANCH
+          echo "CF_BRANCH=${CF_BRANCH}" >> $GITHUB_ENV
+        shell: bash
```

```suggestion
      - name: Update deploy branch for merged PRs
        if: needs.event_types.outputs.pr_merged == 'true'
        run: |
            echo "CF_BRANCH=${{ github.event.repository.default_branch }}" >> $GITHUB_ENV
```

Then you can just put a job-level default value for env:
```
  env:
     CF_BRANCH: github.event.pull_request.head.ref
```